--- a/third_party/xorg-server/os/access.c
+++ b/third_party/xorg-server/os/access.c
@@ -1783,6 +1783,7 @@ siCheckAddr(const char *addrString, int length)
     return len;
 }
 
+#if defined(TCPCONN)
 /***
  * Hostname server-interpreted host type
  *
@@ -2027,6 +2028,7 @@ siIPv6CheckAddr(const char *addrString, int length, void *typePriv)
     return len;
 }
 #endif                          /* IPv6 */
+#endif                          /* TCPCONN */
 
 #if !defined(NO_LOCAL_CLIENT_CRED)
 /***
@@ -2160,10 +2162,12 @@ siLocalCredCheckAddr(const char *addrString, int length, void *typePriv)
 static void
 siTypesInitialize(void)
 {
+#if defined(TCPCONN)
     siTypeAdd("hostname", siHostnameAddrMatch, siHostnameCheckAddr, NULL);
 #if defined(IPv6) && defined(AF_INET6)
     siTypeAdd("ipv6", siIPv6AddrMatch, siIPv6CheckAddr, NULL);
 #endif
+#endif
 #if !defined(NO_LOCAL_CLIENT_CRED)
     siTypeAdd("localuser", siLocalCredAddrMatch, siLocalCredCheckAddr,
               &siLocalUserPriv);
