--- a/third_party/xorg-server/hw/vfb/InitInput.c
+++ b/third_party/xorg-server/hw/vfb/InitInput.c
@@ -133,9 +133,9 @@ InitInput(int argc, char *argv[])
     k = AddInputDevice(serverClient, vfbKeybdProc, TRUE);
     xiclass = MakeAtom(XI_MOUSE, sizeof(XI_MOUSE) - 1, TRUE);
-    AssignTypeAndName(p, xiclass, "Xvfb mouse");
+    AssignTypeAndName(p, xiclass, "Xmin mouse");
     xiclass = MakeAtom(XI_KEYBOARD, sizeof(XI_KEYBOARD) - 1, TRUE);
-    AssignTypeAndName(k, xiclass, "Xvfb keyboard");
+    AssignTypeAndName(k, xiclass, "Xmin keyboard");
     (void) mieqInit();
 }
 
