diff --git a/third_party/xorg-server/os/utils.c b/third_party/xorg-server/os/utils.c
--- a/third_party/xorg-server/os/utils.c
+++ b/third_party/xorg-server/os/utils.c
@@ -187,7 +187,7 @@ Bool noGEExtension = FALSE;
 
 Bool CoreDump;
 
-Bool enableIndirectGLX = FALSE;
+Bool enableIndirectGLX = TRUE;
 
 Bool AllowByteSwappedClients = TRUE;
 
@@ -543,7 +543,8 @@ UseMsg(void)
     ErrorF("-fakescreenfps #       fake screen default fps (1-600)\n");
     ErrorF("-fp string             default font path\n");
     ErrorF("-help                  prints message with these options\n");
     ErrorF("+iglx                  Allow creating indirect GLX contexts\n");
-    ErrorF("-iglx                  Prohibit creating indirect GLX contexts (default)\n");
+    ErrorF("-iglx                  Prohibit creating indirect GLX contexts\n");
+    ErrorF("                       (indirect GLX is enabled by default in Xmin)\n");
     ErrorF("-I                     ignore all remaining arguments\n");
 #ifdef RLIMIT_DATA
