diff --git a/third_party/xorg-server/fb/fbbits.h b/third_party/xorg-server/fb/fbbits.h
--- a/third_party/xorg-server/fb/fbbits.h
+++ b/third_party/xorg-server/fb/fbbits.h
@@ -661,7 +661,7 @@ POLYLINE(DrawablePtr pDrawable,
                 }
                 e = -len;
                 e1 <<= 1;
-                e3 = e << 1;
+                e3 = e * 2;
                 FIXUP_ERROR(e, octant, bias);
                 if (and == 0) {
                     while (len--) {
@@ -820,7 +820,7 @@ POLYSEGMENT(DrawablePtr pDrawable,
                 }
                 e = -len;
                 e1 <<= 1;
-                e3 = e << 1;
+                e3 = e * 2;
                 FIXUP_ERROR(e, octant, bias);
                 if (!capNotLast)
                     len++;
