--- a/third_party/libXfont2/src/fontfile/fontfile.c
+++ b/third_party/libXfont2/src/fontfile/fontfile.c
@@ -69,6 +69,7 @@ static int FontFileOpenBitmapNCF (FontPathElementPtr fpe, FontPtr *pFont,
 				  fsBitmapFormatMask fmask,
 				  FontPtr non_cachable_font);
 
+#ifndef XMIN_BUILTIN_FONTS_ONLY
 int
 FontFileNameCheck (const char *name)
 {
@@ -137,6 +138,7 @@ FontFileFreeFPE (FontPathElementPtr fpe)
     FontFileFreeDir ((FontDirectoryPtr) fpe->private);
     return Successful;
 }
+#endif
 
 static int
 transfer_values_to_alias(char *entryname, int entrynamelength,
@@ -1113,6 +1115,7 @@ FontFileListNextFontOrAlias(pointer client, FontPathElementPtr fpe,
     return ret;
 }
 
+#ifndef XMIN_BUILTIN_FONTS_ONLY
 static const xfont2_fpe_funcs_rec fontfile_fpe_funcs = {
 	.version = XFONT2_FPE_FUNCS_VERSION,
 	.name_check = FontFileNameCheck,
@@ -1137,3 +1140,4 @@ FontFileRegisterLocalFpeFunctions (void)
 {
     register_fpe_funcs(&fontfile_fpe_funcs);
 }
+#endif
